### PR TITLE
Bjorncs/vespa enforcer extensions

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa-enforcer-extensions/src/main/java/com/yahoo/vespa/maven/plugin/enforcer/DependencyEnforcer.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import java.util.regex.Pattern;
+
+/**
+ * Enforces that all expected dependencies are present.
+ * Fails by default for rules that do not match any dependencies.
+ * Similar to the built-in banned-dependency rule in maven-enforcer-plugin.
+ *
+ * @author bjorncs
+ */
+public class DependencyEnforcer implements EnforcerRule {
```
Comment (gjoranv) [inline vespa-enforcer-extensions/src/main/java/com/yahoo/vespa/maven/plugin/enforcer/DependencyEnforcer.java:37]: Suggest to rename to _EnforceDependencies_ for consistency with most existing enforcer rules.. Rule name should also be _enforceDependencies_. Where is the actual rule name defined, is it just a non-capitalized version of the class name?

**Review by gjoranv — CHANGES_REQUESTED** [review]

File: vespa-enforcer-extensions/src/main/java/com/yahoo/vespa/maven/plugin/enforcer/DependencyEnforcer.java (hunk 1 above)
Comment (bjorncs) [reply vespa-enforcer-extensions/src/main/java/com/yahoo/vespa/maven/plugin/enforcer/DependencyEnforcer.java]: Yes, there is some reflection magic under the hood. That is how injection of the `allowed` list is done.

**Comment by bjorncs** [comment]
@gjoranv PTAL

**Review by gjoranv — APPROVED** [review]
This is great!

**Pull request merged**